topics/homework_topic4_Lobashova/OOP.py

```diff
@@ -0,0 +1,81 @@
+# Write a TestCase class with methods for setup(), run(), and teardown().
+# Create objects of the TestCase class to represent individual test cases.
+class TestCase:
+    # Constructor of the class
+    def __init__(self, name):
+        self.name = name  # Attribute/Property/State
+
+    def setup(self):
+        print(f"Preparing for: {self.name}")
+
+    def run(self):
+        print(f"Running test case: {self.name}")
+
+    def teardown(self):
+        print(f"Clearing after: {self.name}")
+
+
+print('For first test')
+test = TestCase("Login Test")
+test.setup()
+test.run()
+test.teardown()
+
+print('For second test')
+test2 = TestCase("Index page Test")
+test2.setup()
+test2.run()
+test2.teardown()
+
+
+# Implement method overriding in a test automation context.
+# Override a method in the child test class to customize the test execution.
+class ChildTestCase(TestCase):
+
+    def setup(self):
+        print(f"Child preparing for: {self.name}")
+
+    def run(self):
+        print(f"Child running test case: {self.name}")
+
+
+print('For child test')
+child_test = ChildTestCase("Child Login Test")
+child_test.setup()
+child_test.run()
+child_test.teardown()  # will use method from parent class
+
+
+# Create a multiple inheritance example: Write a class that inherits from multiple parent classes
+# (e.g., BaseTest and a custom mixin class), and check how MRO impacts method calls.
+class BaseTest:
+    # Constructor of the class
+    def __init__(self, name):
+        self.name = name  # Attribute/Property/State
+
+    def setup(self):
+        print(f"Base preparing for: {self.name}")
+
+    def run(self):
+        print(f"Base running test case: {self.name}")
+
+    def teardown(self):
+        print(f"Base clearing after: {self.name}")
+
+
+class MixTest(TestCase, BaseTest):
```

Comment:
A "mixin class" is not a "mix" of classes. Despite there being no errors in your code, it is important to understand what "mixin classes" are. Here are a few resources:
- https://www.linkedin.com/pulse/understanding-mixins-python-powerful-tool-code-joseph-kalapurackal/
- https://www.pythontutorial.net/python-oop/python-mixin/
- https://dev.to/bikramjeetsingh/write-composable-reusable-python-classes-using-mixins-6lj